examples/experimental/i3s-17-and-debug/tile-debug.js

```diff
@@ -61,3 +65,117 @@ function getChildrenInfo(children) {
+  checkGeometryVsTexture(tileWarnings, tile);
+
+  return tileWarnings;
+}
+
+function checkBoundingVolumes(tileWarnings, tile) {
+  const boundingType = getBoundingType(tile);
+
+  switch (boundingType) {
+    case OBB: {
+      validateObb(tileWarnings, tile);
+      break;
+    }
+    case MBS: {
+      validateMbs(tileWarnings, tile);
+      break;
+    }
+    default:
+      console.warn('Validator - Not supported Bounding Volume Type'); //eslint-disable-line
+  }
+}
+
+function validateObb(tileWarnings, tile) {
+  if (!tile.parent) {
+    return;
+  }
+
+  const parentObb = getParentObb(tile);
+  const tileVertices = getTileObbVertices(tile);
+  const isTileObbInsideParentObb = isAllObbVerticesInsideParentObb(parentObb, tileVertices);
+
+  if (isTileObbInsideParentObb) {
+    return;
+  }
+
+  const title = `OBB of ${tile.id} tile doesn't fit into ${tile.parent.id} parent tile OBB`;
+  tileWarnings.push({type: BOUNDING_VOLUME_WARNING_TYPE, title});
+}
+
+function validateMbs(tileWarnings, tile) {
+  // TODO MBS Validation
+  return;
+}
+
+// LOD spec https://github.com/Esri/i3s-spec/blob/master/format/LevelofDetail.md
+function checkLOD(tileWarnings, tile) {
+  const parentLod = tile.parent && tile.parent.lodMetricValue;
+
+  if (!parentLod || tile.lodMetricValue > parentLod) {
+    return;
+  }
+
+  const title = `${tile.id} tile LOD ${tile.lodMetricValue} < parent ${tile.parent.id} tile ${
+    tile.parent.lodMetricValue
+  } LOD`;
+  tileWarnings.push({type: LOD_WARNING_TYPE, title});
+}
+
+function checkGeometryVsTexture(tileWarnings, tile) {
+  // TODO Geometry vs Texture Validation
+  return;
+}
+
+function getParentObb(tile) {
+  const {center, halfSize, quaternion} = tile.parent.header.obb;
+
+  return new OrientedBoundingBox().fromCenterHalfSizeQuaternion(center, halfSize, quaternion);
+}
+// TODO check if Obb generates properly
+function getTileObbVertices(tile) {
+  const geometry = new CubeGeometry();
+  const halfSize = tile.header.obb.halfSize;
+  const {attributes} = geometry;
+  const positions = new Float32Array(attributes.POSITION.value);
+  const obbCenterCartesian = Ellipsoid.WGS84.cartographicToCartesian(tile.header.obb.center);
+
+  let vertices = [];
+
+  for (let i = 0; i < positions.length; i += 3) {
+    const positionsVector = new Vector3(
+      (positions[i] *= halfSize[0]),
+      (positions[i + 1] *= halfSize[1]),
+      (positions[i + 2] *= halfSize[2])
+    );
+    const rotatedPositions = positionsVector
+      .transformByQuaternion(tile.header.obb.quaternion)
+      .add(obbCenterCartesian);
+    vertices = vertices.concat(rotatedPositions);
+  }
+
+  return vertices;
+}
+
+function isAllObbVerticesInsideParentObb(parentObb, tilePositions) {
+  let isTileObbInsideParent = true;
+
+  for (let index = 0; index < tilePositions.length / 3; index += 3) {
+    const point = [tilePositions[index], tilePositions[index + 1], tilePositions[index + 2]];
+    const distance = parentObb.distanceTo(point);
```

Comment:
Does this `distanceTo` consume absolute cartesian (ECEF) coordinates or cartographic instead? You have them here in the `point` vector.

Reply:
Added point transformation from `ECEF -> Cartographic ` before measuring distance.